File: CS-Examples/10_Conversion/ToPdfWithPassword.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;

namespace ToPdfWithPassword
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new Document instance
			Document document = new Document();

			// Load the Word document from the specified file path
			document.LoadFromFile(@"..\..\..\..\..\..\..\Data\ConvertedTemplate.docx");

			// Create a ToPdfParameterList instance to configure PDF conversion options
			ToPdfParameterList toPdf = new ToPdfParameterList();

			// Set a password for the PDF encryption
			string password = "E-iceblue";
			toPdf.PdfSecurity.Encrypt(password, password, PdfPermissionsFlags.Default, PdfEncryptionKeySize.Key128Bit);

			// Save the document as a PDF file with encryption and the specified output file name
			document.SaveToFile("EncryptWithPassword.pdf", toPdf);

			// Dispose the Document object after use
			document.Dispose();

            //view the PDF file.
            WordDocViewer("EncryptWithPassword.pdf");
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB-Examples/10_Conversion/ToPdfWithPassword.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Doc

Namespace ToPdfWithPassword
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new Document object
			Dim document As New Document()

			' Load a Word document from the specified file path
			document.LoadFromFile("..\..\..\..\..\..\..\Data\ConvertedTemplate.docx")

			' Create a new ToPdfParameterList object to set parameters for PDF conversion
			Dim toPdf As New ToPdfParameterList()

			' Specify the password for encrypting the PDF
			Dim password As String = "E-iceblue"

			' Set PDF security options to encrypt the PDF with the specified password and permissions
			toPdf.PdfSecurity.Encrypt(password, password, Spire.Doc.PdfPermissionsFlags.Default, Spire.Doc.PdfEncryptionKeySize.Key128Bit)

			' Save the document to a PDF file with the specified encryption settings
			document.SaveToFile("EncryptWithPassword.pdf", toPdf)

			' Dispose of the Document object to release resources
			document.Dispose()


			'view the PDF file.
			WordDocViewer("EncryptWithPassword.pdf")
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
